ops/internal/config/superchain.go

```diff
@@ -18,12 +18,7 @@ const (
 )
 
 func ParseSuperchain(in string) (Superchain, error) {
-	switch Superchain(in) {
-	case MainnetSuperchain, SepoliaSuperchain, SepoliaDev0Superchain:
-		return Superchain(in), nil
-	default:
-		return "", fmt.Errorf("unknown superchain: '%s'", in)
-	}
+	return Superchain(in), nil
```

Comment:
This is a bit smelly, having a string enum that we can just cast any string to. Might be better to just rip out the string enum.

Reply:
Yes this will never error because the validation has been removed. So I would say we need to remove this function now and move towards using a `string` (instead of string alias) as the superchain type. Then if we use the `init()` function and populate a `var ValidSuperchains map[string]config.SuperchainDefinition` by reading from every `superchain/configs` dir, then we can query that map to determine if a given superchain is valid instead of calling the `ParseSuperchain` function

Reply:
`Superchain` is actually not an alias presently but a distinct type with string as its underlying type. In [c2f83c1](https://github.com/ethereum-optimism/superchain-registry/pull/1011/commits/c2f83c1e9815f16ff5ceb8855f73bc0c1973ee50) I changed it so it _is_ an alias (meaning, just another way of saying `string`). See this article https://go.dev/blog/alias-names. It makes the refactor much simpler to achieve.

Reply:
I didn't put in the `ValidSuperchains` mapping yet because we already have `paths.Superchains()` which returns a slice of strings with the existing superchains (computed at runtime). This is used in a bunch of places and I think does the job we need here.